### PR TITLE
[14005] Using content filter API to perform content filtering

### PR DESCRIPTION
**Pull request by MiguelCompany**
Just what the title says.

Additionally, I added a LogConsumer that shows errors on the `Output` tab.

### CODE REVIEW

File: include/eprosimashapesdemo/shapesdemo/ShapeSubscriber.h [hunk 1]
```diff
@@ -34,6 +34,7 @@
 #include <fastdds/dds/subscriber/qos/DataReaderQos.hpp>
 #include <fastdds/dds/subscriber/Subscriber.hpp>
 #include <fastdds/dds/topic/Topic.hpp>
+#include <fastdds/dds/topic/ContentFilteredTopic.hpp>
```
Comment (JLBuenoLopez-eProsima) [inline include/eprosimashapesdemo/shapesdemo/ShapeSubscriber.h:37]: Please, sort alphabetically

**Review by JLBuenoLopez-eProsima — COMMENTED** [review]
Just a nitpick. Overall LGTM!

File: include/eprosimashapesdemo/shapesdemo/ShapeSubscriber.h (hunk 1 above)
Comment (MiguelCompany) [reply include/eprosimashapesdemo/shapesdemo/ShapeSubscriber.h:36]: done

**Review by JLBuenoLopez-eProsima — APPROVED** [review]
LGTM!

**Pull request merged**